### PR TITLE
[WIP] `rake rails:update` & Rails 5 upgrade

### PR DESCRIPTION
**Pull request by tristanil**
* `rake rails:update`

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,38 +1,26 @@
 require File.expand_path('../boot', __FILE__)
 
 require 'rails/all'
-require 'csv'
```
Comment (drusepth) [inline config/application.rb]: We should make sure CSV exports still work.

File: config/environments/production.rb [hunk 2]
```diff
@@ -1,101 +1,79 @@
...
-  config.serve_static_files = true
+  # Enable Rack::Cache to put a simple HTTP cache in front of your application
+  # Add `rack-cache` to your Gemfile before enabling this.
+  # For large-scale production use, consider using a caching reverse proxy like
+  # NGINX, varnish or squid.
+  # config.action_dispatch.rack_cache = true
+
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  config.serve_static_files = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (drusepth) [inline config/environments/production.rb]: We should make sure our static error pages in the other repo still display.

File: config/initializers/cookies_serializer.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # Be sure to restart your server when you modify this file.
 
-Rails.application.config.action_dispatch.cookies_serializer = :hybrid
```
Comment (tristanil) [inline config/initializers/cookies_serializer.rb]: This change might log out all existing users.

File: config/initializers/session_store.rb [hunk 4]
```diff
@@ -1,10 +1,3 @@
...
 
-Notebook::Application
-  .config.session_store :cookie_store,
-                        key: '_plan_characters_session'
-
-# Use the database for sessions instead of the cookie-based default,
-# which shouldn't be used to store highly confidential information
-# (create the session table with "rails generate session_migration")
-# Notebook::Application.config.session_store :active_record_store
+Rails.application.config.session_store :cookie_store, key: '_notebook_session'
```
Comment (tristanil) [inline config/initializers/session_store.rb]: This change will definitely log out all existing users.

**Comment by drusepth** [comment]
🚀

**Pull request merged**